### PR TITLE
greatest common denominator substep

### PR DESCRIPTION
**Pull request by ldworkin**
https://github.com/socraticorg/mathsteps/issues/41

### CODE REVIEW

File: lib/simplifyExpression/fractionsSearch/divideByGCD.js [hunk 1]
```diff
@@ -40,6 +42,16 @@ function divideByGCD(fraction) {
...
 
+  const intermediateNumerator = Node.Creator.operator(
+    '/', [Node.Creator.constant(numeratorValue), Node.Creator.constant(gcd)]);
+  const intermediateDenominator = Node.Creator.operator(
+    '/', [Node.Creator.constant(denominatorValue), Node.Creator.constant(gcd)]);
+  const intermediateFraction = Node.Creator.operator(
+    '/', [intermediateNumerator, intermediateDenominator]);
+  const status = Node.Status.nodeChanged(
+    ChangeTypes.FIND_GCD, fraction, intermediateFraction);
+  substeps.push(status);
```
Comment (ldworkin) [inline lib/simplifyExpression/fractionsSearch/divideByGCD.js]: Usually we call `resetChangeGroups` after this but I'm not sure if that's needed here?

File: lib/ChangeTypes.js [hunk 2]
```diff
@@ -80,6 +80,8 @@ module.exports = {
   SIMPLIFY_FRACTION: 'SIMPLIFY_FRACTION',
   // e.g. 2/-3 -> -2/3
   SIMPLIFY_SIGNS: 'SIMPLIFY_SIGNS',
+  // e.g. 15/6 -> (5*3)/(2*3)
+  FIND_GCD: 'FIND_GCD',
```
Comment (evykassirer) [inline lib/ChangeTypes.js]: pedagogy!! 🎉

File: test/simplifyExpression/fractionsSearch/divideByGCD.test.js [hunk 3]
```diff
@@ -15,5 +19,19 @@ describe('simplifyFraction', function() {
...
     ['-1/-1', '1'],
   ];
-  tests.forEach(t => testdivideByGCD(t[0], t[1]));
+  tests.forEach(t => testDivideByGCD(t[0], t[1]));
+});
+
+describe('simplifyFraction', function() {
+  const tests = [
+    ['15/6',
+     ['15/3 / (6/3)'],
```
Comment (evykassirer) [inline test/simplifyExpression/fractionsSearch/divideByGCD.test.js]: I think the numerator should have parens around it too (if you keep this version of the rule)

File: lib/ChangeTypes.js (hunk 2 above)
Comment (evykassirer) [inline lib/ChangeTypes.js]: in your test (and the code you wrote) you'll get `15/3 / (6/3)` and not `(5*3)/(2*3)`

I think the latter is cleaner (nested fractions render more funky in the app) - what do you think is better?

File: lib/simplifyExpression/fractionsSearch/divideByGCD.js (hunk 1 above)
Comment (evykassirer) [inline lib/simplifyExpression/fractionsSearch/divideByGCD.js]: ah, cause `fraction` and `intermediateFraction` aren't at all part of the nodes used in the next step? Yeah, it's probably fine then

changeGroups are kinda messy in general heh and not well tested so I wouldn't worry too much about it for now (but thinking about them a bit when you can is still good)

File: lib/ChangeTypes.js (hunk 2 above)
Comment (ldworkin) [inline lib/ChangeTypes.js]: Ha good call -- I agree, changed!

**Pull request merged**